CsvParser/csvParser.go

```diff
@@ -94,47 +98,158 @@ func CsvParser(filePath string, numIncidents ...int) []incident.DefaultIncident
 	return defaultIncidents
 }
 
-//* This function takes in an array of 100 default incidents because that is what is provided by the csvParser above *//
-func convertDefaultIncidentToIncident(defaultIncidents []incident.DefaultIncident) []incident.Incident {
-	incidentArr := make([]incident.Incident, len(defaultIncidents), len(defaultIncidents))
-	for i := range defaultIncidents {
-		incidentArr[i] = &defaultIncidents[i]
-	}
-	return incidentArr
+//convert a default incident that implements our interface to an incident object designed to exist in a json file
+//It returns incident of type IncidentData to 
+func convertDefaultIncidentToIncidentData(i incident.DefaultIncident) incident.IncidentData {
+	gpStart, gpEnd := i.GetGoodPeriod()
+	bpStart, bpEnd := i.GetBadPeriod()
+	gMetric := i.GetGoodMetric()
+	bMetric := i.GetBadMetric()
+	severity := i.GetSeverity()
+	testsAffected := i.GetTestsAffected()
+	gpInfo := i.GetGoodPeriodInfo()
+	bpInfo := i.GetBadPeriodInfo()
+	iInfo := i.GetIncidentInfo()
+	locationString := i.GetLocation()
+	asn := i.GetASN()
+	inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, asn, locationString, severity, testsAffected, gpInfo, bpInfo, iInfo}
+	return inc
 }
 
-func makeJsonObjFile(arr []incident.Incident) *os.File {
-	// numObjects determines how many incidents are stored in the json
-	const numObjects = 1
-	f, err := os.Create("incidents.json")
-	var objs [numObjects]incident.IncidentData
-	if err != nil {
-		log.Fatal(err)
-		return f
+//Takes in a location code and breaks it down into different location levels
+//Returns an array of string, where each string represents a location
+func parseLocationCode(locationCode string) []string {
+	locationCodesArr := make([]string, 0)
+
+	if (len(locationCode)) > 6 {
+		for i := 0; i < 6; i = i + 2 {
+			locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+		}
+
+		locationCodesArr = append(locationCodesArr, locationCode[6:])
```

Comment:
What is this locationCode[6:] element?

Reply:
perhaps add variables with more descriptive names that hold values like locationCode[6:] to make code more readable :)